packages/ibm-products/src/components/ConditionBuilder/ConditionGroupBuilder/ConditionGroupBuilder.js

```diff
@@ -74,46 +123,181 @@ const ConditionGroupBuilder = ({
     );
   };
 
-  const handleFocusOnClose = (evt) => {
-    let previousClose = evt.currentTarget
-      ?.closest('[role="row"]')
-      ?.parentElement?.previousSibling?.querySelector(
-        '[data-name="closeCondition"]'
-      );
+  const showConditionSubGroupPreviewHandler = (conditionIndex) => {
+    setShowConditionSubGroupPreview(conditionIndex);
+  };
+  const hideConditionSubGroupPreviewHandler = () => {
+    setShowConditionSubGroupPreview(-1);
+  };
+  const showConditionPreviewHandler = (conditionIndex) => {
+    setShowConditionPreview(conditionIndex);
+  };
+  const hideConditionPreviewHandler = () => {
+    setShowConditionPreview(-1);
+  };
+  const onStatementChangeHandler = (updatedStatement) => {
+    onChange({
+      ...group,
+      statement: updatedStatement,
+    });
+  };
 
-    if (previousClose) {
-      previousClose.focus();
-    }
+  const onConnectorOperatorChange = (op) => {
+    onChange({
+      ...group,
+      groupOperator: op,
+    });
   };
-  return (
-    <div className={`${className}  eachGroup`}>
+
+  const getSentenceVariant = () => {
```

Comment:
in order to cut back on file length and improve readability i would move this and `getTreeVariant` into their own separate component files.

Reply:
Hey,
That makes sense and thought in the same way at first. But if you notice , most of the functionalities and handler definitions are same for both variant, except some of the JSX structure. So I thought separating into own files will make it more complex and duplicate as we need to passdown all handlers and call back. 
What do you think?

Reply:
ok this should be fine for now. let's keep an eye on it